controller/aks-cluster-config-handler.go

```diff
@@ -709,16 +712,32 @@ func (h *Handler) updateUpstreamClusterState(ctx context.Context, secretsCache w
 
 	// check tags for update
 	if config.Spec.Tags != nil {
+
 		if !reflect.DeepEqual(config.Spec.Tags, upstreamSpec.Tags) {
 			logrus.Infof("Updating tags for cluster [%s]", config.Spec.ClusterName)
 			tags := containerservice.TagsObject{
 				Tags: *to.StringMapPtr(config.Spec.Tags),
 			}
-			_, err = resourceClusterClient.UpdateTags(ctx, config.Spec.ResourceGroup, config.Spec.ClusterName, tags)
+			response, err := resourceClusterClient.UpdateTags(ctx, config.Spec.ResourceGroup, config.Spec.ClusterName, tags)
 			if err != nil {
 				return config, err
 			}
-			return h.enqueueUpdate(config)
+
+			// Azure may have a policy that automatically adds upstream default tags to a cluster resource. We don't
+			// have a good way to detect that policy. We handle this case by checking if Azure returns an unexpected
+			// state for the tags and if so, log the response and move on. Any upstream tags regenerated on the cluster
+			// by Azure will be synced back to rancher.
+			managedCluster := containerservice.ManagedCluster{}
+			err = retry.OnError(retry.DefaultBackoff, func(err error) bool { return strings.HasSuffix(err.Error(), "asynchronous operation has not completed") }, func() error {
+				managedCluster, err = response.Result(*resourceClusterClient)
+				return err
+			})
+
+			if !reflect.DeepEqual(tags, managedCluster.Tags) && response.Response().StatusCode == http.StatusOK {
+				logrus.Infof("Tags were not updated as expected for cluster %s, expected %s, actual %s, moving on", to.StringMap(tags.Tags),to.StringMap(managedCluster.Tags), config.Spec.ClusterName)
```

Comment:
change the order of arguments
```suggestion
				logrus.Infof("Tags were not updated as expected for cluster %s, expected %s, actual %s, moving on", config.Spec.ClusterName, to.StringMap(tags.Tags),to.StringMap(managedCluster.Tags))
```